server/jamovi/server/instance.py

```diff
@@ -870,6 +870,27 @@ def _on_module(self, request):
             except Exception as e:
                 log.exception(e)
                 self._coms.send_error(str(e), None, self._instance_id, request)
+        elif request.command == jcoms.ModuleRR.ModuleCommand.Value('SHOW'):
+            self._set_module_visibility(request.name, True)
+            self._coms.send(None, self._instance_id, request)
+            self._session.notify_global_changes()
+        elif request.command == jcoms.ModuleRR.ModuleCommand.Value('HIDE'):
+            self._set_module_visibility(request.name, False)
+            self._coms.send(None, self._instance_id, request)
+            self._session.notify_global_changes()
+
+    def _set_module_visibility(self, name, value):
+        modules = Modules.instance()
+        if modules.set_visibility(name, value):
+            settings = Settings.retrieve('modules')
+            hidden_mods = settings.get('hidden', [ ])
+            if value:
+                while name in hidden_mods:
```

Comment:
not a big deal (because this will always be a really short list), but a more efficient approach would be:

```
hidden_mods = [ mod for mod in hidden_mods if mod != name ]
```

Reply:
which lines is this replacing?

Reply:
replaces

```
while name in hidden_mods:
    hidden_mods.remove(name)
```

Reply:
you've got a few other options:

```
hidden_mods = list(filter(lambda x: x != name, hidden_mods))
```

```
hidden_mods = set(hidden_mods)
hidden_mods.remove(name)
hidden_mods = list(hidden_mods)
```

these all have the advantage of only iterating over the list once.